src/server/DeviceServer.js

```diff
@@ -295,48 +295,45 @@ class DeviceServer {
 
           this._devicesById.set(deviceID, device);
 
-          device.completeProtocolInitialization();
+          const systemInformation = await device.completeProtocolInitialization();
+          const { uuid: appHash } = FirmwareManager.getAppModule(systemInformation);
 
           const existingAttributes =
             await this._deviceAttributeRepository.getByID(deviceID);
-          const ownerID = existingAttributes && existingAttributes.ownerID;
 
-          // ip newer in the current device state, than in
-          // existing attributes from repo, so filter them,
-          // and update only what we need.
-          // eslint-disable-next-line no-unused-vars
-          const { ip, ...attributes } = existingAttributes || {};
-          device.updateAttributes(attributes);
-          device.setStatus(DEVICE_STATUS_MAP.GOT_REPO_ATTRIBUTES);
-
-          this.publishSpecialEvent(
-            SYSTEM_EVENT_NAMES.SPARK_STATUS,
-            'online',
-            deviceID,
+          const {
+            claimCode,
+            currentBuildTarget,
+            imei,
```

Comment:
from where we actually get info like imei, isCellural, last_iccid ? Seems now its just always undefined.